### ISSUE TITLE
startup autocmd event documentation is inconsistent with how it behaves

### ISSUE THREAD

**Issue by errael** [neovim/neovim#31069]
### Problem

Started putting together a simple doc PR, things got more complicated.

Initially:
```
diff --git a/src/nvim/vvars.lua b/src/nvim/vvars.lua
--- a/src/nvim/vvars.lua
+++ b/src/nvim/vvars.lua
@@ -173,2 +173,3 @@
         chan             |channel-id|
+        info             As returned by |nvim_get_chan_info()|.
         cmdlevel         Level of cmdline.
```

Then saw some related cleanup in autocmd.txt, for `ChanInfo` and `ChanOpen`
(but perhaps it's only a matter of style)

A key of `chan_info` instead of `info` seems to make more sense. (breaking change)

In `autocmd.txt` removing the following for `ChanInfo` and `ChanOpen` is more consistent in docs.
```
        See |nvim_get_chan_info()| for the format of
        the info Dictionary.
```
and then could (or not) mention value format with the key like
```
				Sets these |v:event| keys:
                                    chan_info   as from |nvim_get_chan_info()|
```

-----

_Back to the primary issue_

The particular: frequently there is no `ChanInfo` event.
Not sure if it's a doc/code/spec/cockpit issue.

In develop.txt there's (twice)
```
- Call |nvim_set_client_info()| after connecting, so users and plugins can
  detect the UI by handling the |ChanInfo| event. This avoids the need for ...
```
But with `nvim-tui` the `ChanInfo` event is intermittent. With both `neovide`
and `goneovim` the event never happens.

At https://neovim.io/doc/user/starting.html#--embed there's
```
--embed ... Waits for the ... nvim_ui_attach() before sourcing startup files
... The client can do other requests before nvim_ui_attach (e.g. nvim_get_api_info
```

`ChanOpen` event is never seen. Might be my specific usage/environment.

In `autocmd.txt`
```
UIEnter
    After a UI connects via |nvim_ui_attach()|,
    or after builtin TUI is started.
```
```
ChanInfo
    State of channel changed, for instance the
    client of a RPC channel described itself.
```

### Steps to reproduce

Observations using the script below

With `nvim-tui` either
1. `UIEnter` without a later `ChanInfo`
   In `UIEnter` handler `nvim_get_chan_info()` `client` value is populated.
2. `UIEnter` followed by `ChanInfo`
   In `UIEnter` handler `nvim_get_chan_info()` `client` value is empty dict.
   In `ChanInfo` `vim.v.event.info` is populated as expected.

With `neovide` never see `ChanInfo` 
1. In `UIEnter` handler `nvim_get_chan_info()` value is populated.

```lua
vim.api.nvim_create_autocmd("ChanInfo", {
    callback = function(ev)
        vim.print(string.format("ChanInfo: ev: %s", vim.inspect(ev, {newline=""})))
        vim.print(string.format("ChanInfo: v:event info: %s", vim.inspect(vim.v.event.info)))
    end
})

vim.api.nvim_create_autocmd("UIEnter", {
    callback = function(ev)
        vim.print(string.format("UIEnter: ev: %s", vim.inspect(ev, {newline=""})))
        vim.print(string.format("UIEnter: v:event: %s", vim.inspect(vim.v.event, {newline=""})))
        vim.print(string.format("UIEnter: chan info: %s", vim.inspect(
            vim.api.nvim_get_chan_info(vim.v.event.chan), {newline=""})))
    end
})
```

### Expected behavior

More accurate descriptions

(@bfredl follow up on #31057)

### Nvim version (nvim -v)

https://github.com/neovim/neovim/commit/0da4d89558a05fb86186253e778510cfd859caea

### Vim (not Nvim) behaves the same?

NA

### Operating system/version

ubuntu

### Terminal name/version

gnome-terminal

### $TERM environment variable

xterm-256color

### Installation

make install

**Comment by errael** [comment]
@justinmk @zeertzjq This issue is fundamentally about `ChanInfo` docs. The first part of the OP is about a documentation cleanup PR for ChanInfo that could go so far as to include a breaking change (rename v:event key to `chan_info`  from `info` (too ambiguous)). I can do that PR with some guidance. The 2nd half of OP is about how `ChanInfo` behavior is described; probably more for future reference.

**Comment by justinmk** [comment]
> The name of the `v:event` key for `ChanOpen`/`ChanInfo`, `info`, seems too ambiguous;

? v:event is context-dependent. The meaning of the keys is explained in ChanInfo docs. Not ambiguous.

Is there anything else besides the docs update needed here? Hard to follow the description here.

**Comment by errael** [comment]
(Yeah, TL;DR)

This is the crux of it. 

> In develop.txt there's (twice)
> 
> ```
> - Call |nvim_set_client_info()| after connecting, so users and plugins can
>   detect the UI by handling the |ChanInfo| event. This avoids the need for ...
> ```

ChanInfo does not always occur. User/plugin can't reliably use it as described.

> Not sure if it's a doc/code/spec/cockpit issue.

As mentioned in the following, I wonder if doing things before nvim_ui_attach() contributes to this.

> At https://neovim.io/doc/user/starting.html#--embed there's
> 
> ```
> --embed ... Waits for the ... nvim_ui_attach() before sourcing startup files
> ... The client can do other requests before nvim_ui_attach (e.g. nvim_get_api_info
> ```

Observations. Note in the following, when there is no `ChanInfo` event, the "info" is available in the UIEnter event with `nvim_get_chan_info()`.

> With `nvim-tui` either
> 
>     1. `UIEnter` without a later `ChanInfo`
>     2. `UIEnter` followed by `ChanInfo`
> 
> With `neovide` never see `ChanInfo`

**Comment by errael** [comment]
Forgot to mention that `goneovim` does never generate `ChanInfo` event and never has populated `client` data. Was thinking of this issue as something that might be part of future cleanup/rework/trashcan. Reminds me of #31052.
